tests/codex/stores/testsqlitestore.nim

```diff
@@ -0,0 +1,150 @@
+import std/options
+import std/os
+import std/sets
+
+import pkg/asynctest
+import pkg/chronos
+import pkg/stew/byteutils
+
+import pkg/codex/chunker
+import pkg/codex/blocktype as bt
+import pkg/codex/stores
+
+import ../helpers
+
+suite "SQLite Store":
+  var
+    store: SQLiteStore
+
+  let
+    newBlock = bt.Block.new("New Block".toBytes()).expect("should not fail")
+    repoDir = getAppDir() / "repo"
+
+  setup:
+    removeDir(repoDir)
+    require(not dirExists(repoDir))
+    store = SQLiteStore.new(repoDir)
```

Comment:
I don't think that the `SQLiteStore` should be messing around with creating directories, I would move that out.

Reply:
`SQLiteStore` does not create a directory, at least not directly; rather, that's handle by nim-datastore's `SQLiteDatastore` constructor:

https://github.com/status-im/nim-datastore/blob/master/datastore/sqlite_datastore.nim#L201-L211

We can change the behavior of nim-datastore, of course, but maybe it can be considered in separate issue/PR?

Reply:
We can address it as part of a separate PR, that should be fine, but this is definitely something to fix.

Reply:
Tracking in https://github.com/status-im/nim-codex/issues/178 and https://github.com/status-im/nim-datastore/issues/21.

Reply:
See https://github.com/status-im/nim-datastore/pull/23.